### DISCUSSION TITLE
Imputation yes/no is not displayed when nimpute is higher than 1

### DISCUSSION THREAD

**Discussion by adgfo** [kogalur/randomForestSRC#130] [Q&A] [answered]
Hi!
I have set "train <- rfsrc(y~., data, ntree=1000, nsplit=10, importance='anti', na.action='na.impute', nimpute=1, seed=7)". Printing "train" indicates "was data imputed: yes". Checking "imputed.data" indeed indicates a dataframe of imputed values.
However, when nimpute=2, or higher number, "was data imputed: yes" is not indicated. Checking "imputed.data" is NULL, the dataframe of imputed values is empty.
Any reason for why this happens?
Thank you!

**Comment by ishwaran** [comment, marked as answer]
From the help file of` rfsrc`

 _The missing data algorithm can be iterated by setting
          ‘nimpute’ to a positive integer greater than 1.  When
          iterated, at the completion of each iteration, missing data
          is imputed using OOB non-missing terminal node data which is
          then used as input to grow a new forest.  A side effect of
          iteration is that missing values in the returned objects
          ‘xvar’, ‘yvar’ are replaced by imputed values.  In other
          words the incoming data is overlaid with the missing data._

Thus, when `nimpute` is greater than 1, the grow forest is treated as if there was no missing data.  In particular, due to overlaying imputed data, `$xvar `and `$yvar` contain no missing values.  We handle it this way for various reasons, for example when prediction is requested using `predict.rfsrc`.
